explorer/main.cpp

```diff
@@ -30,18 +31,33 @@ namespace cl = llvm::cl;
 namespace path = llvm::sys::path;
 
 auto ExplorerMain(int argc, char** argv, void* static_for_main_addr,
-                  llvm::StringRef relative_prelude_path) -> int {
+                  llvm::StringRef relative_prelude_path,
+                  llvm::raw_ostream& out_stream, llvm::raw_ostream& err_stream)
+    -> int {
   llvm::setBugReportMsg(
-
       "Please report issues to "
       "https://github.com/carbon-language/carbon-lang/issues and include the "
       "crash backtrace.\n");
   llvm::InitLLVM init_llvm(argc, argv);
 
   // Printing to stderr should flush stdout. This is most noticeable when stderr
   // is piped to stdout.
-  llvm::errs().tie(&llvm::outs());
+  err_stream.tie(&out_stream);
```

Comment:
Per the header, this variant of ExplorerMain is intended for binary use. It's used by both //explorer and //installer:carbon, both of which need the tie behavior. So moving it out of this would only result in the behavior being copied into the two callers.

Reply:
Hm, OK. Do we need to pass in the output and error streams? I think it's the fact that they're passed in and not hardcoded to `outs` / `errs` that's making me wonder whether this is the right place for this call. Both callers pass in `llvm::outs` and `llvm::errs`.

Reply:
Will do.